src/js/angular/security/controllers.js

```diff
@@ -104,6 +104,37 @@ const parseAuthorities = function (authorities) {
     };
 };
 
+const compactAuthorities = function (authorities) {
+    const writeAllRepos = authorities.find(authority => authority === `${WRITE_REPO_PREFIX}*`);
+    if (writeAllRepos) {
+        return  _.remove(authorities, function (_authority) {
+            if (_authority.indexOf(READ_REPO_PREFIX) === -1 && _authority.indexOf(WRITE_REPO_PREFIX) === -1) {
```

Comment:
Can't we have constants for these WRITE_REPO_PREFIX_* and READ_REPO_PREFIX_*? As far as I can see, these are just static strings and can be defined once and reused. 
E.g. WRITE_ALL_REPO_WILDCARD or something.